ui/src/app/views/workspace/component.ts

```diff
@@ -32,7 +35,22 @@ export class WorkspaceComponent implements OnInit {
     this.cohortsService.getCohortsInWorkspace('123', '123').subscribe(
           cohortsReceived => {
             this.cohortList = cohortsReceived;
+            this.CohortEditService.list().then(
+              cohorts => {
+                for (const coho of cohorts) {
+                  this.cohortList.push(coho);
+                }
+              }
+            );
           });
     this.currentUrl = this.router.url;
   }
+
+  addCohort(): void {
+    this.router.navigate([this.currentUrl + '/cohort/' + this.CohortEditService.add()]);
```

Comment:
I'd suggest two reasons for waiting to assign an id: 1) not all cohorts will be saved, so we would need to "clean up" non saved cohorts that had been allocated from the server (unreliable) or add a separate "assign ID" endpoint (extra work, and a state management issue), and 2) it requires a round trip to the server before the router navigation effect can occur (introduces lag in the ux).